examples/dmd/wave_equation.cpp

```diff
@@ -0,0 +1,632 @@
+        sout.open(vishost, visport);
+        if (!sout)
+        {
+            cout << "Unable to connect to GLVis server at "
+                 << vishost << ':' << visport << endl;
+            visualization = false;
+            cout << "GLVis visualization disabled.\n";
+        }
+        else
+        {
+            sout.precision(precision);
+            sout << "solution\n" << *mesh << dudt_gf;
+            sout << flush;
+        }
+    }
+
+    StopWatch fom_timer, dmd_training_timer, dmd_prediction_timer;
+    fom_timer.Start();
+
+    // 8. Perform time-integration (looping over the time iterations, ti, with a
+    //    time-step dt).
+    ode_solver->Init(oper);
+    double t = 0.0;
+
+    vector<double> ts;
+
+    fom_timer.Stop();
+
+    dmd_training_timer.Start();
+
+    int curr_window = 0;
+    vector<CAROM::DMD*> dmd_u;
+    dmd_u.push_back(new CAROM::DMD(u.Size(), dt));
+    dmd_u[curr_window]->takeSample(u, t);
+    ts.push_back(t);
+
+
+    dmd_training_timer.Stop();
+
+
+    bool last_step = false;
+    for (int ti = 1; !last_step; ti++)
+    {
+        fom_timer.Start();
+        if (t + dt >= t_final - dt/2)
+        {
+            last_step = true;
+        }
+
+        ode_solver->Step(u, dudt, t, dt);
+        fom_timer.Stop();
+
+        dmd_training_timer.Start();
+        dmd_u[curr_window]->takeSample(u, t);
+
+        if (last_step || (ti % windowNumSamples) == 0)
+        {
+            cout << "step " << ti << ", t = " << t << endl;
+
+            if(rdim != -1) {
+                cout << "Creating DMD with rdim: " << rdim << endl;
+                dmd_u[curr_window]->train(rdim);
+            }
+
+            else {
+                cout << "Creating DMD with energy fraction: " << ef << endl;
+                dmd_u[curr_window]->train(ef);
+            }
+
+            if(!last_step) {
+                curr_window++;
+                dmd_u.push_back(new CAROM::DMD(u.Size(), dt));
+                dmd_u[curr_window]->takeSample(u, t);
+            }
+
+        }
+        ts.push_back(t);
+        dmd_training_timer.Stop();
+
+        if (last_step || (ti % vis_steps) == 0)
+        {
+
+            u_gf.SetFromTrueDofs(u);
+            dudt_gf.SetFromTrueDofs(dudt);
+
+            if (visualization)
+            {
+                sout << "solution\n" << *mesh << u_gf << flush;
+            }
+
+            if (visit)
+            {
+                visit_dc.SetCycle(ti);
+                visit_dc.SetTime(t);
+                visit_dc.Save();
+            }
+        }
+        oper.SetParameters(u);
+
+    }
+
+    // 9. Save the final solution. This output can be viewed later using GLVis:
+    //    "glvis -m wave_equation.mesh -g wave_equation-final.gf".
+    {
+        ofstream osol("wave_equation-final.gf");
+        osol.precision(precision);
+        u_gf.Save(osol);
+        dudt_gf.Save(osol);
+    }
+
+    // 10. Calculate the DMD modes.
+
+
+    Vector true_solution_u(u.Size());
```

Comment:
This can be removed, simply using `u` rather than copying it.

Reply:
Fixed.